src/sentry/integrations/grpc/services/scm_service.py

```diff
@@ -0,0 +1,641 @@
+                for oi in org_integrations:
+                    integration = integration_service.get_integration(
+                        integration_id=oi.integration_id
+                    )
+                    if integration and integration.provider == provider_key:
+                        org_integration = oi
+                        break
+                if not org_integration:
+                    raise ValueError(f"No integration found for provider {provider_key}")
+            else:
+                # Get first available SCM integration
+                org_integrations = integration_service.get_organization_integrations(
+                    organization_id=org.id
+                )
+                # Find first SCM integration
+                org_integration = None
+                scm_providers = [
+                    "github",
+                    "gitlab",
+                    "bitbucket",
+                    "bitbucket_server",
+                    "vsts",
+                    "github_enterprise",
+                ]
+                for oi in org_integrations:
+                    integration = integration_service.get_integration(
+                        integration_id=oi.integration_id
+                    )
+                    if integration and integration.provider in scm_providers:
+                        org_integration = oi
+                        break
+                if not org_integration:
+                    raise ValueError(
+                        f"No SCM integration found for organization {request.organization_id}"
+                    )
+
+            # Get the full integration object
+            integration = integration_service.get_integration(
+                integration_id=org_integration.integration_id
+            )
+            if not integration:
+                raise ValueError(f"Integration {org_integration.integration_id} not found")
+
+            # Get integration class and create installation
+            integration_cls = integrations_manager.get(integration.provider)
+            if not integration_cls:
+                raise ValueError(f"Integration provider {integration.provider} not found")
+
+            # Create a mock integration model with minimal required fields for get_installation
+            # This avoids silo mode issues
+            class MockIntegration:
+                def __init__(self, id, provider, name, external_id, metadata):
+                    self.id = id
+                    self.provider = provider
+                    self.name = name
+                    self.external_id = external_id
+                    self.metadata = metadata
+
+            integration_model = MockIntegration(
+                id=integration.id,
+                provider=integration.provider,
+                name=integration.name,
+                external_id=integration.external_id,
+                metadata=integration.metadata or {},
+            )
```

Comment:
Any idea what this is intended to solve? I'm also not even sure which silo this is intended to operate in.